src/transformers/loss_utils.py

```diff
@@ -0,0 +1,98 @@
+import torch
+import torch.nn as nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from .models.detr.loss_detr import ForObjectDetectionLoss, ForSegmentationLoss
+
+
+def DefaultCrossEntropyLoss(logits, labels, **kwargs):
+    # Upcast to float if we need to compute the loss to avoid potential precision issues
+    logits = logits.float()
+    # Shift so that tokens < n predict n
+    shift_logits = logits[..., :-1, :].contiguous()
+    shift_labels = labels[..., 1:].contiguous()
+
+    # Flatten the tokens
+    shift_logits = shift_logits.view(-1, kwargs["vocab_size"])
+    shift_labels = shift_labels.view(-1)
+    # Enable model parallelism
+    shift_labels = shift_labels.to(shift_logits.device)
+
+    num_items = kwargs.pop("num_items", None)
+
+    if num_items is not None:
+        # Calculate the CrossEntropyLoss manually when using grad accum
+        log_probs = nn.functional.log_softmax(shift_logits, dim=-1)
+        loss = -log_probs[range(shift_labels.size(0)), shift_labels]
+        loss = loss.sum() / num_items
+    else:
+        loss = nn.functional.cross_entropy(shift_logits, shift_labels, ignore_index=-100)
```

Comment:
```suggestion
    if num_items is not None:
        loss = nn.functional.cross_entropy(shift_logits, shift_labels, ignore_index=-100, reduction="sum")
        loss = loss / num_items
    else:
        loss = nn.functional.cross_entropy(shift_logits, shift_labels, ignore_index=-100)
```

Reply:
@muellerzr as you wrote the snippet!

Reply:
i believe currently it is not ignoring the losses and potentially the functional API might be faster?
```
loss.masked_fill_(shift_labels == -100, 0)
```

Reply:
aligned with the suggestion here!